libraries/FSTools/FSTools.cpp

```diff
@@ -0,0 +1,247 @@
+bool FSTools::attemptToMountFS(fs::FS & fs)
+{
+  LittleFSConfig littleFSCfg(false);
+  SPIFFSConfig SPIFFSCfg(false); 
+  //  try to apply the "safe" no format config to the FS... doesn't matter which.. just need one to apply.. 
+  if (!fs.setConfig(littleFSCfg) && ! fs.setConfig(SPIFFSCfg) )
+  {
+  	return false; 
+  }
+  return fs.begin(); 
+}
+
+
+bool FSTools::mountAlternativeFS(  FST::FS_t type, const FST::layout & layout, bool keepMounted )
+{
+	FSConfig * pCfg{nullptr};
+    LittleFSConfig littleFSCfg(false);
+    SPIFFSConfig SPIFFSCfg(false); 
+    reset();
+
+	switch (type) {
+		case FST::SPIFFS : {
+			_pFS.reset( new FS(FSImplPtr(new spiffs_impl::SPIFFSImpl    (_getStartAddr(layout) , _getSize(layout) , layout.page, layout.block, 5)))  );
+			pCfg = &SPIFFSCfg; 
+			break;
+		}
+		case FST::LITTLEFS : {
+			_pFS.reset( new FS(FSImplPtr(new littlefs_impl::LittleFSImpl(_getStartAddr(layout) , _getSize(layout) , layout.page, layout.block, 5)))  );
+			pCfg = &littleFSCfg; 
+			break;
+		}
+	};
+
+	if (_pFS && pCfg && _pFS->setConfig(*pCfg) && _pFS->begin()) {
+		if (!keepMounted) {
+			_pFS->end();
+		}
+		_mounted = true; 
+		_layout = &layout;
+		return true; 
+	}
+
+	if (_pFS) {
+		_pFS.reset();
+	}
+	_mounted = false; 
+	return false; 
+
+}; 
+
+
+bool FSTools::mounted()
+{
+	return _mounted; 
+};
+
+
+
+
+bool FSTools::moveFS(fs::FS & destinationFS)
+{
+    uint32_t sourceFileCount = 0; 
+    uint32_t sourceByteTotal = 0; 
+    bool result = false; 
+
+	//  do not init new fs... until old one is copied to test... 
+	if (!_mounted || !_pFS) {
+		//Serial.println("Source FS not mounted"); 
+		return false; 
+	}
+
+
+  	uint32_t startSector = (ESP.getSketchSize() + FLASH_SECTOR_SIZE - 1) & (~(FLASH_SECTOR_SIZE - 1)); 
+  	//uint32_t endSector = (uint32_t)&_FS_start - FST::lowestSPIFFSstartAddr;
+  	uint32_t lowestFSStart = 0x40300000;
+  	
+  	 if (_layout) {
+  	 	lowestFSStart = _layout->startAddr;  	
+  		//Serial.printf("**  _layout->startADDR = 0x%08x\n", _layout->startAddr ); 
+  	 }
+  	
+  	uint32_t endSector = lowestFSStart - 0x40200000;  
+  	uint32_t tempFSsize = endSector - startSector;
+
+  	//Serial.printf("TempFS:  start: %u, end: %u, size: %u, sketchSize = %u, _FS_start = %u\n", startSector, endSector, tempFSsize, ESP.getSketchSize(), (uint32_t)&_FS_start );  
+
+    fileListIterator(*_pFS, "/", [&sourceFileCount, &sourceByteTotal, this](File & f) {
+      if (f) {
+        sourceFileCount++;
+        sourceByteTotal += f.size(); 
+        //_dumpFileInfo(f); 
+    }
+    });
+
+    //Serial.printf("%u Files Found Total Size = %u\n", sourceFileCount, sourceByteTotal); 
+    //Serial.printf("Size of dummy FS = %u\n", tempFSsize); 
+
+    FS tempFS = FS(FSImplPtr(new littlefs_impl::LittleFSImpl(startSector, tempFSsize, FS_PHYS_PAGE, FS_PHYS_BLOCK, 5)));
+
+     if (tempFS.format() && tempFS.begin()) {
+     	 if (_copyFS(*_pFS,tempFS)) {
+     	 	//Serial.println("Files copied to temp File System"); 
+     	 	reset(); 
```

Comment:
General comment, something fishy is going on with tabs and spaces.  It's a mix of them now and it messes up block alignment pretty bad if they're not identical (i.e. like in the GH web GUI).  Can you please change tabs->space in your editor for these files to clean it up and make it easier to read?

Reply:
I'd add `libraries/FSTools` in `tests/restyle.sh` (variable `all`) and run the script.

Reply:
I'd just worked that out, I've done it, and confirmed on the CL that both files were updated.